### ISSUE TITLE
handle multiline patterns in lambdas

### ISSUE THREAD

**Issue by avh4** [avh4/elm-format#305]
Deal with `pleaseReport "TODO" "multiline pattern in lambda"` in `ElmFormat/Render/Box.hs`.  Add test cases and implement formatting for this.

Example code that triggers this:
```
f = \{x, --A
  y} -> ()
```

**Comment by avh4** [comment]
The formatting I came up with is

```elm
lambdaWithMultilinePattern =
    \{ x
     , --A
       y
     }
    ->
        ()
```

Since no one has ever reported this before, that seems reasonable to use without too much additional though for now.

**Comment by avh4** [comment]
Fixed in c18714f7...0f180c1f
